intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,184 @@
+        .map(trainingPhrase -> buildTrainingPhraseFromEncodedString(trainingPhrase))
+        .collect(Collectors.toList()); 
+    trainingPhrasesProtobufList.addAll(userTrainingPhrases);
+    intentProtobufBuilder.addAllTrainingPhrases(trainingPhrasesProtobufList);
+    return intentProtobufBuilder;
+  }
+
+  private static Intent.Builder addResponses(UseCase.Intent intent,
+      Intent.Builder intentProtobufBuilder) {
+    final Intent.Message message = Intent.Message.newBuilder()
+        .setText(Intent.Message.Text.newBuilder().addAllText(intent.getResponsesList()).build())
+        .build();
+    return intentProtobufBuilder.addMessages(message);
+  }
+
+  private static Intent.Builder addOutputContexts(UseCase.Intent intent,
+      Intent.Builder intentProtobufBuilder, String projectID) {
+    List<Context> outputContextProtobufList= new ArrayList<Context>();
+    for(String successor: intent.getSuccessorsList()) {
+      Context contextProtobuf = Context.newBuilder()
+      .setLifespanCount(1)
+      .setName("projects/" + projectID + "/agent/sessions/-/contexts/" + successor + "Context")
+      .build();
+      outputContextProtobufList.add(contextProtobuf);
+    }
+    return intentProtobufBuilder.addAllOutputContexts(outputContextProtobufList);
+  }
+
+  private static void createOrUpdateIntents(List<Intent> intentProtobufList,
+    IntentsClient intentsClient, Map<String, String> intentDisplayNameToName,
+    ProjectAgentName parent) {
+    for(Intent intent: intentProtobufList) {
+      if(intentDisplayNameToName.containsKey(intent.getDisplayName())) {
+        intentsClient.updateIntent(intent, "en");
+      }
+      else {
+        intentsClient.createIntent(parent, intent);
+      }
+    }
+  }
+
+  private static Intent buildIntentProtobufForIntent(UseCase.Intent intent, String projectID) {
+    Intent.Builder intentProtobufBuilder = Intent.newBuilder();
+    intentProtobufBuilder.setDisplayName(intent.getIntentName());        
+    // output contexts need to be set to the input contexts of the successor intents
+    addOutputContexts(intent, intentProtobufBuilder, projectID);
+    intentProtobufBuilder.addInputContextNames("projects/" + projectID +
+        "/agent/sessions/-/contexts/" + intent.getIntentName() + "Context");
+
+    intentProtobufBuilder = addTrainingPhrases(intent, intentProtobufBuilder);
+    intentProtobufBuilder = addResponses(intent, intentProtobufBuilder);        
+    intentProtobufBuilder.addAllEvents(intent.getEventsList());
+
+    if(intentDisplayNameToName.containsKey(intent.getIntentName())) {
+      intentProtobufBuilder.setName(intentDisplayNameToName.get(intent.getIntentName()));
+    }
+    return intentProtobufBuilder.build();
+  }
+
+  private static Map<String, String> getIntentDisplayNametoNameMapping(
+        IntentsClient intentsClient, ProjectAgentName parent) {
+    Map<String, String> intentDisplayNameToName = new HashMap<String, String>();
```

Comment:
Instead of declaring new map here, we can directly update the class level map and rename method to something like generate....(

Reply:
It was not a class variable earlier, so this was building and returning the map. Updated.